src/lib/browse/layers/CriticalIssuesLayerControl.svelte

```diff
@@ -0,0 +1,153 @@
+<script lang="ts">
+  import type { GeoJSONSource, MapGeoJSONFeature } from "maplibre-gl";
+  import { map } from "../../../stores";
+  import { CollapsibleCard, ColorLegend, InteractiveLayer } from "../../common";
+  import {
+    Checkbox,
+    CheckboxGroup,
+    ErrorMessage,
+    FormElement,
+  } from "../../govuk";
+  import {
+    cleanupSource,
+    emptyGeojson,
+    overwriteCircleLayer,
+  } from "../../maplibre";
+  import { parseCriticalIssuesExcel } from "../data";
+
+  let fileInput: HTMLInputElement;
+
+  let source = "criticals";
+  let color = "red";
+  let show = true;
+  let numberIssues = 0;
+  let errorMessage = "";
+
+  cleanupSource($map, source);
+  $map.addSource(source, {
+    type: "geojson",
+    data: emptyGeojson(),
+    cluster: true,
+    clusterMaxZoom: 14,
+    clusterRadius: 50,
+  });
+
+  // Render clusters
+  overwriteCircleLayer($map, {
+    id: `${source}-clusters`,
+    source,
+    filter: ["has", "point_count"],
+    color,
+    opacity: 0.9,
+    // 20 pixels for clusters with < 15 points, 30 pixels for < 30 points, and 40 pixels for more
+    radius: ["step", ["get", "point_count"], 20, 15, 30, 30, 40],
+  });
+  $map.addLayer({
+    id: `${source}-counts`,
+    source,
+    type: "symbol",
+    filter: ["has", "point_count"],
+    layout: {
+      "text-field": "{point_count_abbreviated}",
+    },
+  });
+
+  // Render individual criticals
+  overwriteCircleLayer($map, {
+    id: `${source}-points`,
+    source,
+    filter: ["!", ["has", "point_count"]],
+    color,
+    opacity: 0.9,
+    radius: 15,
+    strokeColor: "black",
+    strokeWidth: 3,
+  });
+
+  // InteractiveLayers don't manage one layer
+  $: {
+    if ($map.getLayer(`${source}-counts`)) {
+      $map.setLayoutProperty(
+        `${source}-counts`,
+        "visibility",
+        show ? "visible" : "none"
+      );
+    }
+  }
+
+  async function onChange(e: Event) {
+    try {
+      let gj = await parseCriticalIssuesExcel(fileInput.files![0]);
+      ($map.getSource(source) as GeoJSONSource).setData(gj);
+      numberIssues = gj.features.length;
+      errorMessage = "";
+    } catch (err) {
+      errorMessage = `The file you loaded is broken: ${err}`;
+    }
+  }
+
+  function tooltip(feature: MapGeoJSONFeature): string {
+    let x = `<div style="max-width: 30vw;">`;
```

Comment:
Inconsistent use of bold/underline on the value / label for the value. Style suggestions welcome!